### PR TITLE
Add UI for starring and unstarring a challenge.

### LINKED ISSUE #1202: Add UI for starring and unstarring a challenge.
**Expected behaviour:** Add a button in challenge page for `starring` and `unstarring` a challenge with count.

### PR DESCRIPTION
**Pull request by RishabhJain2018**
fixes #1202 . @deshraj @aka-jain Please review the PR.
star-demo

### CODE REVIEW

**Comment by deshraj** [comment]
Please add a star icon on the left side of the button.

File: frontend/src/js/controllers/challengeCtrl.js [hunk 1]
```diff
@@ -1036,6 +1036,53 @@
...
 
+        // Get the stars count and user specific starred or unstarred
+        parameters = {};
+        parameters.url = "challenges/"+ vm.challengeId + "/";
+        parameters.method = 'GET';
+        parameters.token = userKey;
+        parameters.callback = {
+            onSuccess: function(response) {
+                var details = response.data;
+                console.log(details);
```
Comment (deshraj) [inline frontend/src/js/controllers/challengeCtrl.js]: Please remove this.

**Comment by deshraj** [comment]
Can you please paste a screenshot again?

**Comment by RishabhJain2018** [comment]
@deshraj Please take a look!
shot1
shot2

**Pull request merged**